src/ptf/dataplane.py

```diff
@@ -694,6 +703,107 @@ def packets(self, device, port=None):
             pkt, time = queue.pop(0)
             yield (rcv_port, pkt, time)
 
+    PollResult = namedtuple('PollResult', ['device', 'port', 'packet', 'time'])
+    """
+    A base class used for the result of poll(). No matter what kind of
+    additional information subclasses include, for backwards compatibility
+    callers must be able to unpack them as if they were a tuple with this
+    format. The pair of the @device number and the @port number indicate where
+    the packet was received, @packet is the packet itself, and @time is the time
+    it was received.
+    """
+
+    # On success, poll() returns a PollSuccess object which contains the device
+    # number and port number on which a matching packet was found, the packet
+    # itself, and the time at which it was received.
+    class PollSuccess(PollResult):
+        """ Returned by poll() when it successfully finds a matching packet. """
+        def __new__(cls, device, port, packet, expected_packet, time):
```

Comment:
Could you add a comment explaining that `__new__` is used (and we cannot use `__init__`) because we inherit from `namedtuple` which is immutable?